doctr/models/recognition/master/pytorch.py

```diff
@@ -171,14 +182,11 @@ def forward(
             gt, seq_len = torch.from_numpy(_gt).to(dtype=torch.long), torch.tensor(_seq_len)
             gt, seq_len = gt.to(x.device), seq_len.to(x.device)
 
-        if self.training:
-            if target is None:
-                raise AssertionError("In training mode, you need to pass a value to 'target'")
-            tgt_mask = self.make_mask(gt)
+            # Compute source mask and target mask
+            source_mask, target_mask = self.make_source_and_target_mask(encoded, gt)
```

Comment:
the source_mask is needed whatever the mode (training vs inference) but the target_mask is only needed in training right?

We need to avoid forgetting cases (being in training mode != being passed a target), especially for the evaluation loops for our reference scripts (the model will be in `self.training = False` but it will be given a target so that it can compute the loss)

Reply:
decoder only:
train:
- we need the combined target mask to ensure the model cannot 'cheat'
- source mask is optional
inference:
- normally we don't need to pass any mask while inference ... but (MASTER is weird :sweat_smile:) .. without the masks the model will not work correctly .. i can't explain 100% why but what i think is, that we also while inference need the masking to force the models predictions 'step by step' otherwise we end up in something like this: 'aaaaabbbb'

Will update this :+1: (also crnn / sar pytorch)